tests/span_tests.cpp

```diff
@@ -483,6 +483,66 @@ SUITE(span_tests)
 #endif
     }
 
+    TEST(from_unique_pointer_construction)
+    {
+        {
+            auto arr = new int[4];
+            auto ptr = std::unique_ptr<int>(arr);
```

Comment:
Doesn't this result in the wrong `delete` being called when `ptr` goes out of scope?